### ISSUE TITLE
Cluster analysis on faces?

### ISSUE THREAD

**Issue by taewookim** [davidsandberg/facenet#608]
Is it possible to do a cluster analysis on L2 distance of image embeddings? (for example, to determine which face is most often seen with another face?)

**Comment by MaartenBloemen** [comment]
Yes you can do that with the DBSCAN algorithm. For an example see [cluster.py](https://github.com/davidsandberg/facenet/blob/master/contributed/cluster.py).
